File: GitHubRepositoriesSearch/Models/Gitrepoinfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

// Data object same as in client side
namespace GitHubRepositoriesSearch.Models
{
    public class Gitrepoinfo
    {
        public Gitrepoinfo()
        {

        }

        // repository name
        public string name { get; set; }

        // repository url
        public string repoUrl { get; set; }

        // owner avatar
        public string avatar { get; set; }
    }
}

File: GitHubRepositoriesSearch/src/app/models/gitrepoinfo.ts
// Data object same as in server side
export class Gitrepoinfo {
    // repository name
    name: string;
    // repository url
    repoUrl: string;
    // owner avatar
    avatar: string;
}

export class Repositories {
    total_count: number;
    incomplete_results: boolean;
    items: Gitrepoinfo[];
}

